examples/gno.land/p/demo/int256/int256.gno

```diff
@@ -1,64 +1,83 @@
-// This package provides a 256-bit signed integer type, Int, and associated functions.
 package int256
 
 import (
+	"errors"
+
 	"gno.land/p/demo/uint256"
 )
 
-var one = uint256.NewUint(1)
+var (
+	int1  = NewInt(1)
+	uint0 = uint256.NewUint(0)
+	uint1 = uint256.NewUint(1)
+)
 
 type Int struct {
-	abs *uint256.Uint
-	neg bool
+	value uint256.Uint
 }
 
-// Zero returns a new Int set to 0.
-func Zero() *Int {
-	return NewInt(0)
+// New creates and returns a new Int initialized to zero.
+func New() *Int {
+	return &Int{}
 }
 
-// One returns a new Int set to 1.
-func One() *Int {
-	return NewInt(1)
+// NewInt allocates and returns a new Int set to the value of the provided int64.
+func NewInt(x int64) *Int {
+	return New().SetInt64(x)
 }
 
-// Sign returns:
+// Zero returns a new Int initialized to 0.
 //
-//	-1 if x <  0
-//	 0 if x == 0
-//	+1 if x >  0
-func (z *Int) Sign() int {
-	z.initiateAbs()
+// This function is useful for creating a starting point for calculations or
+// when an explicit zero value is needed.
+func Zero() *Int { return &Int{} }
+
+// One returns a new Int initialized to one.
+//
+// This function is convenient for operations that require a unit value,
+// such as incrementing or serving as an identity element in multiplication.
+func One() *Int { return int1 }
```

Comment:
Shouldn't this return a pointer to a copy?

Reply:
fixed

https://github.com/gnolang/gno/pull/2846/commits/2ced1989a92a05df4f519a518eb48b8e5b6cb5c5